st3/sublime_lib/settings_listener.py

```diff
@@ -52,12 +52,33 @@ def _on_settings_changed(self) -> None:
 
 
 class GlobalSettingsListener(BaseSettingsListener, sublime_plugin.EventListener):
+    """
+    A subclass of :class:`sublime_plugin.EventListener`
+    that also listens for changes in a global settings object.
+
+    Subclasses must set the `SETTINGS_NAME` class variable to the name of the global settings.
+
+    Example usage:
+
+    .. code-block:: python
+        from sublime_lib import GlobalSettingsListener, on_setting_changed
+
+        class JsCustomConfigurationsListener(GlobalSettingsListener)
+            SETTINGS_NAME = 'JS Custom.sublime-settings'
+
+            @on_setting_changed('configurations')
+            def configurations_changed(self, new_configuration, old_configuration):
+                if self.settings.get('auto_build', False):
+                    sublime.active_window().run_command('build_js_custom_syntaxes')
+    """
     SETTINGS_NAME = ''
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(sublime.load_settings(self.SETTINGS_NAME), *args, **kwargs)
```

Comment:
I'd like to have some verification here that `self.SETTINGS_NAME` has been overridden in the subclass because that's easy to miss.

Reply:
I agree, though I'm not sure about the mechanism. Should we raise an error? Log a warning? Either way, it might be tricky to test.

We'll also want a special case for when someone unthinkingly exports `GlobalSettingsListener ` itself and Sublime instantiates it.

Reply:
I'd expect an exception, probably a `RuntimeError`.

And true, we need something to prevent the imported class from interfacing with ST.

Reply:
Added.